numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -621,6 +594,31 @@ static PyObject *
 /**end repeat**/
 
 static PyObject *
+voidtype_str(PyObject *self){
+    PyObject *arr, *ret = NULL;
+
+    if (!PyDataType_HASFIELDS(((PyVoidScalarObject*)self)->descr)) {
+        PyObject  *item, *item_str;
+
+        item = gentype_generic_method(self, NULL, NULL, "item");
+        if (item == NULL) {
+            return NULL;
+        }
+
+        item_str = PyObject_Str(item);
+        Py_DECREF(item);
+        return item_str;
+    }
+
+    arr = PyArray_FromScalar(self, NULL);
+    if (arr != NULL) {
+        ret = PyObject_Str((PyObject *)arr);
+        Py_DECREF(arr);
+    }
+    return ret;
+}
```

Comment:
Can you bundle the removal of `gentype_str` and `gentype_repr`, and the addition of `voidtype_str` into a single commit (and preferably, PR)? Or does that change by itself cause breakage?